Documentation/kube-aws-approver.md

```diff
@@ -0,0 +1,118 @@
+- There exists an `instance` in `running state` corresponding to the `nodename` from the CertificateRequest `CommonName`
+- A `node` with the `nodename` from CertificateRequest CommonName is `part` of the cluster and in `Ready` state.
+- The `instance` should belong to an AutoScalingGroup.
+- The AutoSaclingGroup belongs to the list of whitelisted AutoScalingGroups provided to `kube-aws-approver` at startup.
+
+Only when a CertificateSigningRequest from a node looking to renew its client certificate meets `all` the above requirements, the CertificateSigningRequest is `Approved` [more](#certificateSigningRequest-approval-process)
+
+### Criteria 2: A new node trying to join the cluster requests for a client certificate.
+
+- The CertificateRequest `Organization` is equal to `system:nodes`
+- The CertificateRequest `DNSNames` list is empty
+- The CertificateRequest `EmailAddresses` list is empty
+- The CertificateRequest `IPAddresses` list is empty
+- The CertificateRequest `CommonName` has `system:node:` prefix
+- The CertificateSigningRequest `Groups` has `system:bootstrappers`
+- There exists an instance in `running state` corresponding to the `nodename` from the  CertificateRequest `CommonName`
+- The CertificateSigningRequest `Username` shoudd be of the form `system:bootstrappers:<aws-instance-id>` [How you might achieve this?](aws-authenticator)
+- The `instance-id` extracted from CertificateSigningRequest `Username` matches the `instance-id` extracted from AWS API for the `nodename` in CertificateRequest `CommonName`
+- The node with `nodename` from CertificateRequest CommonName is `not part` of the cluster.
+- The `instance` should belong to an AutoScalingGroup.
+- The AutoSaclingGroup belongs to the list of whitelisted AutoScalingGroups provided to `kube-aws-approver` at startup.
+
+Only when a CertificateSigningRequest from a new node looking for its client certificate meets `all` the above requirements, the CertificateSigningRequest is `Approved` [more](#certificateSigningRequest-approval-process)
+
+`kube-aws-approver` only approves a CertificateSigningRequest that meets one of these two criteria. It never explicitly declines any CertificateSigningRequest.
+The approved CertificateSigningRequest is then signed by the kube-controller-manager and can be used by the node as client certificate.
+
+### CertificateSigningRequest Approval process
+
+The following example go code shows how the `kube-aws-approver` marks a CertificateSigningRequest approved.
+```go
+...
+...
+// csr is the CertificateSigningRequest(https://godoc.org/k8s.io/api/certificates/v1beta1#CertificateSigningRequest) object received by the approver.
+csr.Status.Conditions = append(csr.Status.Conditions, CertificateSigningRequestCondition{
+    Type:    CertificateApproved,
+    Reason:  "AutoApproved",
+    Message: successMessage,
+})
+_, err = client.CertificatesV1beta1().CertificateSigningRequests().UpdateApproval(csr)
+
+...
+...
+```
+
+## How to run?
```

Comment:
"How to run" should probably include copy and pastable examples. See something like https://github.com/coreos/matchbox/blob/master/Documentation/bootkube.md

This just describes the flags. Maybe drop it for now?